provision/aws/rosa_create_cluster.sh

```diff
@@ -5,96 +5,53 @@ if [ -f ./.env ]; then
-  --worker-iam-role arn:aws:iam::${AWS_ACCOUNT}:role/ManagedOpenShift-Worker-Role \
-  --operator-roles-prefix ${CLUSTER_NAME} \
-  --region ${REGION} ${MULTI_AZ_PARAM} ${AVAILABILITY_ZONES_PARAM} \
-  --replicas ${REPLICAS} \
-  --compute-machine-type ${COMPUTE_MACHINE_TYPE} \
-  --machine-cidr ${MACHINE_CIDR} \
-  --service-cidr 172.30.0.0/16 \
-  --pod-cidr 10.128.0.0/14 \
-  --host-prefix 23"
-
-  echo $ROSA_CMD
-  $ROSA_CMD
-fi
-
-mkdir -p "logs/${CLUSTER_NAME}"
-
-function custom_date() {
-    date '+%Y%m%d-%H%M%S'
+function requiredEnv() {
+  for ENV in $@; do
+      if [ -z "${!ENV}" ]; then
+        echo "${ENV} variable must be set"
+        exit 1
+      fi
+  done
 }
 
-echo "Creating operator roles."
-rosa create operator-roles --cluster "${CLUSTER_NAME}" --mode auto --yes > "logs/${CLUSTER_NAME}/$(custom_date)_create-operator-roles.log"
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
-echo "Creating OIDC provider."
-rosa create oidc-provider --cluster "${CLUSTER_NAME}" --mode auto --yes > "logs/${CLUSTER_NAME}/$(custom_date)_create-oidc-provider.log"
+AWS_ACCOUNT=${AWS_ACCOUNT:-$(aws sts get-caller-identity --query "Account" --output text)}
 
-echo "Waiting for cluster installation to finish."
-# There have been failures with 'ERR: Failed to watch logs for cluster ... connection reset by peer' probably because services in the cluster were restarting during the cluster initialization.
-# Those errors don't show an installation problem, and installation will continue asynchronously. Therefore, retry.
-TIMEOUT=$(($(date +%s) + 3600))
-while true ; do
-  if (rosa logs install --cluster "${CLUSTER_NAME}" --watch --tail=1000000 >> "logs/${CLUSTER_NAME}/$(custom_date)_create-cluster.log"); then
-    break
-  fi
-  if (( TIMEOUT < $(date +%s))); then
-    echo "Timeout exceeded"
-    exit 1
-  fi
-  echo "retrying watching logs after failure"
-  sleep 1
-done
+requiredEnv AWS_ACCOUNT COMPUTE_MACHINE_TYPE CLUSTER_NAME REGION REPLICAS VERSION
 
-echo "Cluster installation complete."
-echo
+export CLUSTER_NAME=${CLUSTER_NAME:-$(whoami)}
 
-./rosa_recreate_admin.sh
+echo "Verifying ROSA prerequisites."
+echo "Check if AWS CLI is installed."; aws --version
+echo "Check if ROSA CLI is installed."; rosa version
+echo "Check if ELB service role is enabled."
+if ! aws iam get-role --role-name "AWSServiceRoleForElasticLoadBalancing" --no-cli-pager; then
+  aws iam create-service-linked-role --aws-service-name "elasticloadbalancing.amazonaws.com"
+fi
+rosa whoami
+rosa verify quota
+
+echo "Installing ROSA cluster ${CLUSTER_NAME}"
+
+cd ${SCRIPT_DIR}/../opentofu/modules/rosa/hcp
+tofu init
+tofu workspace new ${CLUSTER_NAME} || true
+export TF_WORKSPACE=${CLUSTER_NAME}
+tofu apply -auto-approve \
+  -var cluster_name=${CLUSTER_NAME} \
+  -var instance_type=${COMPUTE_MACHINE_TYPE} \
+  -var openshift_version=${VERSION} \
+  -var region=${REGION} \
+  -var replicas=${REPLICAS}
```

Comment:
This script used to be idempotent if a following step failed, so we could re-run it via GitHub actions when for example installing of the logging failed. 

I tried to run the following, and even after setting `TF_VAR_token` get an error. 

Ideally it should continue where it stopped. If this is not trivial, at least re-adding the snippet from before would do most of the trick: 

```
if rosa describe cluster --cluster="${CLUSTER_NAME}"; then
  echo "Cluster ${CLUSTER_NAME} already exists."
else
#  ...
``` 

See below for the command I ran, and the output it provided. 

```
tofu workspace select gh-ryan-us-a
tofu apply -var cluster_name=gh-ryan-us-a -var instance_type=m5.xlarge -var openshift_version=4.14.5 -var region=us-east-2 -var replicas=2
``` 

Error received:

```
╷
│ Error: Attempt to index null value
│ 
│   on ../account-roles/main.tf line 7, in locals:
│    7:       policy_details       = data.rhcs_hcp_policies.all_policies.account_role_policies["sts_hcp_installer_permission_policy"]
│     ├────────────────
│     │ data.rhcs_hcp_policies.all_policies.account_role_policies is null
│ 
│ This value is null, so it does not have any indices.
╵
```

Reply:
> I tried to run the following, and even after setting `TF_VAR_token` get an error.

What error are you getting when running `rosa_create_cluster.sh` multiple times?

I previously executed this script multiple times once the cluster had been created (but the later steps failed) and it recognised the tofu state was as expected.

Reply:
Ok, then it might have been my fault as I ran just the tofu command manually. Let's disregard it as you mention you tested that you were able to rerun it.